### ISSUE TITLE
incorrect line number reported for unsupported escape seq errors

### ISSUE THREAD

**Issue by ikappaki** [clj-kondo/clj-kondo#2492]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

v2025.02.20

**platform**

MS-Windows native

**problem**

The linter incorrectly reports the correct row/column number for unsupported escape seq characters (.e.g. `\a`)

**repro**

1. Create a `bad.clj` file with an unsupported escape seq on a line other than 1:
```clj
123
456
"\a"
```
2. Run the linter, the error is incorrectly reported at 1:4, instead of 3:1
```sh
$ clj-kondo --lint bad.clj 
bad.clj:1:4: error: Unsupported escape character: \a.
linting took 6ms, errors: 1, warnings: 0
```
**expected behavior**

The error should be reported at the correct line (line 3).

[ Optional: if you are interested in doing a PR yourself, please leave a note. ]

I'm happy to do the PR, but I need guidance on the best approach.

- The error occurs in `clj-kondo.impl.utils/sexpr` when `expr->tag` calls `(sexpr expr)`, where `expr` is the `rewrite-clj` `token` `"\a"`.
- The exception originates from `clojure.tools.reader`, propagates through `rewrite-clj`, and reaches back to `clj-kondo`.
- `rewrite-clj` appears to lack the correct file line metadata, but `clj-kondo` retains it in the node's metadata suggetsting the fix belongs in `clj-kondo`.

As a solution I can think of, catching `:reader-error` in `sexpr` and reporting it as a new `reader-error` linter is possible but may not be ideal since `sexpr` is widely used in the codebase and would require a `ctx` that may not always be available from the caller.

Below is a FlowStorm stack trace showing where the exception with the wrong line coordinate is thrown, in case it helps.

![Image](https://github.com/user-attachments/assets/b68da81a-d88f-4904-950b-e3b61f32d7fb)
